### PR TITLE
feature: webserver: fusion apps service migration

### PR DESCRIPTION
**Pull request by kjetilhau**
# Description

Migrate the proxy in webserver to the new Fusion Apps API service.

- [x] PR title and description are to the point and understandable
- [x] I have performed a self-review of my own code'

Please select version type the purposed change:
- [x] major
- [ ] minor
- [ ] patch
- [ ] none  

External Relations
- [ ] database migration

## Changeset

- BREAKING CHANGE: The previous proxy endpoint: `/bundles/apps/{appKey}/resources/{*resourcePath}` is no more. Instead use `/fusion-apps/{*wildcard*}` to make proxy calls against the whole Fusion Apps API. See Fusion Apps API Swagger documentation for more information about routes.
- Migrated proxy from Fusion Portal to Fusion Apps API
- Updated packages to latest version
- Refactored and simplified code
- Removed unused Project Portal variables in frontend 
- Update frontend to use the new proxy endpoints
- Make frontend client use the new types provided by fusion core
- Update the legacy app loader to use the new proxy endpoints and and fix types
- Move legacy app code to apps folder for cleaner repository
- Setup build of legacy app loader in build pipeline for constant build an deployment. 
- Update all Fusion dependencies for frontend client

### CODE REVIEW

**Comment by kjetilhau** [comment]
@Noggling Let me know if you have any other suggestions for the proxy url path, currently: `/**fusion-apps**/...` The important this is to have a path that does not interfere with the frontend router.

File: client/apps/legacyAppLoader/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	{ config: { environment: { endpoints: { client: Client; portal: Client } } } }
+> = (configurator, env) => {
+	configurator.configureHttpClient('portal-client', env.env.config?.environment.endpoints.client);
+	configurator.configureHttpClient('portal', env.env.config?.environment.endpoints.portal);
+
+	window['clientBaseUri'] = env.env.config?.environment.endpoints.client.baseUri;
+	// configurator.logger.level = 4;
+	configurator.configureHttpClient('app', {
+		baseUri: new URL('/apps-proxy/', location.origin).href,
+		defaultScopes: ['5a842df8-3238-415d-b168-9f16a6a6031b/.default'],
```
Comment (Noggling) [inline client/apps/legacyAppLoader/src/config.ts:25]: This needs to fixed.  we need this comming from config

File: client/packages/portal-client/src/lib/portal-framework-config.tsx [hunk 2]
```diff
@@ -102,9 +110,13 @@ export function createPortalFramework(portalConfig: PortalConfig) {
...
 
 		enableContext(config);
-		config.configureServiceDiscovery(portalConfig.serviceDiscovery);
 
-		enableAppModule(config, appConfigurator(portalConfig.portalClient.client));
+		enableAppModule(config);
+
+		config.configureHttpClient('app', {
+			baseUri: new URL('/apps-proxy/', location.origin).href,
+			defaultScopes: ['5a842df8-3238-415d-b168-9f16a6a6031b/.default'],
```
Comment (Noggling) [inline client/packages/portal-client/src/lib/portal-framework-config.tsx:118]: This needs to fixed.  we need this comming from config

File: client/packages/portal-client/src/lib/portal-framework-config.tsx [hunk 3]
```diff
@@ -28,13 +29,20 @@ function getClientIdFormScope(scope: string): string | undefined {
 
 export function createPortalFramework(portalConfig: PortalConfig) {
 	return (config: FusionConfigurator) => {
-		config.logger.level = (portalConfig.logger?.level as LoggerLevel) || 0;
+		config.logger.level = 4; // (portalConfig.logger?.level as LoggerLevel) || 0;
```
Comment (Noggling) [inline client/packages/portal-client/src/lib/portal-framework-config.tsx:32]: get form config

File: client/packages/portal-client/src/lib/portal-framework-config.tsx [hunk 4]
```diff
@@ -254,6 +266,11 @@ export function createPortalFramework(portalConfig: PortalConfig) {
 		}
 
 		config.onInitialized<[NavigationModule, TelemetryModule, AppModule, PortalConfigModule]>(async (fusion) => {
+			// const appService = await fusion.serviceDiscovery.resolveService('apps');
```
Comment (Noggling) [inline client/packages/portal-client/src/lib/portal-framework-config.tsx:269]: remove

File: client/packages/core/src/modules/portal-config/types/index.ts [hunk 5]
```diff
@@ -63,7 +63,6 @@ export type AppManifestResponse = {
 };
 
 export type AppManifest = FusionAppManifest & {
-	isPinned?: boolean;
```
Comment (Noggling) [inline client/packages/core/src/modules/portal-config/types/index.ts:66]: find out if this was needed?

File: client/packages/core/src/app/hooks/use-app-loader.ts [hunk 6]
```diff
@@ -68,19 +86,26 @@ export const useAppLoader = (appKey: string) => {
 									},
 								})
 							);
+
+							/** remove app element when application unmounts */
+							// subscription.add(() => appRef.current.remove());
```
Comment (Noggling) [inline client/packages/core/src/app/hooks/use-app-loader.ts:91]: remove

File: client/packages/portal-pages/src/pages/sheared/components/AllApps.tsx [hunk 7]
```diff
@@ -66,7 +67,10 @@ export const AllApps = () => {
 										}
 									);
 								}}
-								onFavorite={(app) => addFavorite(app.key)}
+								onFavorite={(app) => {
+									console.log(app);
```
[the remark this answers is not in the document]
Comment (Noggling) [reply client/packages/portal-pages/src/pages/sheared/components/AllApps.tsx]: jupp

**Review by Noggling — APPROVED** [review]

**Pull request merged**